### DISCUSSION TITLE
Presence vs Postgres Changes

### DISCUSSION THREAD

**Discussion by Sir-Baconn** [supabase/supabase#28853] [Questions]
Hi all,
I am looking for some advice on implementation. I am creating a simple lobby system (in a web app) where there are 2 teams of players. Currently, you can join the lobby and switch teams if desired.

I started this all by using supabase presence for when a player joins/leaves and broadcast for when a player swaps teams. I realized this doesn't work because if player 1 swaps teams and then player 2 joins after the fact, player 2 doesn't know that player 1 swapped teams. I'm assuming I need to incorporate storing who is on what team in the db and listening to postgres changes instead. My questions are:

1. Am I misunderstanding and this is all actually possible just using the presence feature? Should I use a combination of both presence and postgres changes?
2. If I store the state of the teams in the db, how do you go about updating the database when a player leaves (via closing browser tab or window)? Seems difficult since they will be disconnected already

Thanks!

**Comment by GaryAustin1** [comment]
I don't know much about presence, but there is a state you can change... https://supabase.com/docs/guides/realtime/presence?queryGroups=language&language=js#sending-state

Postgres changes would not tell you when a user leaves at all.  Just when a row in the table changes.
